apps/ledger-live-desktop/src/renderer/reducers/accounts.ts

```diff
@@ -67,8 +67,22 @@ export const accountsSelector = (state: { accounts: AccountsState }): Account[]
 
 // NB some components don't need to refresh every time an account is updated, usually it's only
 // when the balance/name/length/starred/swapHistory of accounts changes.
-const accountHash = (a: AccountLike) =>
-  `${a.id}-${a.balance.toString()}-swapHistory(${a.swapHistory?.length || "0"})`;
+const accountHash = (a: AccountLike) => {
+  const baseHash = `${a.id}-${a.balance.toString()}-swapHistory(${a.swapHistory?.length || "0"})`;
+  // Include Canton-specific data in hash to ensure selector detects changes to cantonResources
+  // Without this, when Canton accounts are synced and cantonResources is updated (e.g., instrumentUtxoCounts),
+  // the selector returns stale data because the hash doesn't change, causing components to miss
+  // important Canton-specific data like UTXO counts needed for transaction validation
+  // See: libs/coin-modules/coin-canton/src/bridge/sync.ts
+  const cantonHash =
+    a.type === "Account" &&
+    a.currency.family === "canton" &&
+    "cantonResources" in a &&
+    a.cantonResources
+      ? `-cantonResources(${JSON.stringify(a.cantonResources)})`
+      : "";
+  return baseHash + cantonHash;
```

Comment:
```suggestion
if (a.type === "Account" && a.currency.family === "canton" && "cantonResources" in a) {
 
    return baseHash + cantonHash;
  }
  
  return baseHash;
```

i think it's more readable :)